src/main/java/io/seqera/tower/agent/Agent.java

```diff
@@ -98,19 +101,30 @@ public static void main(String[] args) throws Exception {
     public void run() {
         try {
             validateParameters();
-            checkTower();
-            connectTower();
             sendPeriodicHeartbeat();
-        } catch (Exception e) {
+            infiniteLoop();
+        } catch (Throwable e) {
             logger.error(e.getMessage());
+            if (logger.isTraceEnabled()) {
+                e.printStackTrace();
+            }
```

Comment:
I don't know. I only want to see the stack trace when trace level is enable. What happens if you pass the exception as an argument to `logger.error()`?

Reply:
It prints it in the log. But why avoid that? If there's an expected exception it is more useful to have the full stack trace otherwise in most cases it's impossible to troubleshoot the problem

Reply:
Now, I'm always logging the unknown exceptions.